packages/connection-storage/src/renderer.ts

```diff
@@ -1,12 +1,62 @@
 import { ipcRenderer } from 'hadron-ipc';
+import { EventEmitter } from 'events';
 
 import type { ConnectionStorage as ConnectionStorageMain } from './connection-storage';
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
 export type {
   ConnectionInfo,
   AtlasClusterMetadata,
 } from '@mongodb-js/connection-info';
+import type { ImportConnectionOptions } from './import-export-connection';
+
+export enum ConnectionStorageEvents {
+  ConnectionsChanged = 'connections-changed',
+}
+
+export type ConnectionStorageEventListeners = {
+  [ConnectionStorageEvents.ConnectionsChanged]: () => void;
+};
+
+export class ConnectionStorageBus extends EventEmitter {
+  on<T extends ConnectionStorageEvents>(
+    eventName: T,
+    listener: ConnectionStorageEventListeners[T]
+  ): this {
+    return super.on(eventName, listener);
+  }
+
+  off<T extends ConnectionStorageEvents>(
+    eventName: T,
+    listener: ConnectionStorageEventListeners[T]
+  ): this {
+    return super.off(eventName, listener);
+  }
+
+  once<T extends ConnectionStorageEvents>(
+    eventName: T,
+    listener: ConnectionStorageEventListeners[T]
+  ): this {
+    return super.once(eventName, listener);
+  }
+
+  removeListener<T extends ConnectionStorageEvents>(
+    eventName: T,
+    listener: ConnectionStorageEventListeners[T]
+  ): this {
+    return super.removeListener(eventName, listener);
+  }
+
+  emit<T extends ConnectionStorageEvents>(
+    eventName: T,
+    ...args: Parameters<ConnectionStorageEventListeners[T]>
+  ): boolean {
+    return super.emit(eventName, ...args);
+  }
+}
 
 export class ConnectionStorage {
+  public static events: ConnectionStorageBus = new ConnectionStorageBus();
```

Comment:
The only reason is that we use ConnectionStorage as a static class (that looks like a normal instance when we use it). If I extend the ConnectionStorage with the EventEmitter, I'm not sure it will work properly on the typings side.

Reply:
Yes, just using `extends` keyword will not work here because you are extending the prototype. You can just define the staic methods on the class though, you have to do it for typing purposes on the ConnectionStorageBus class anyway